ecosystem/sep-0006.md

```diff
@@ -276,32 +276,82 @@ Example:
+
+In either case, the JSON message should be identical to the response format for the [/transaction](#single-historical-transaction) endpoint.
+
+As an alternative to using the `callback` parameter, the wallet can poll the transaction endpoint [`/transaction`](#single-historical-transaction) with the request `id` to check the status of the request.
+
+**`jwt` details**
+
+If the anchor's deposit or withdrawal endpoints require [SEP-10](sep-0010.md) authentication, the wallet must include this extra parameter. It allows the anchor to continue the interactive flow without requiring the user to authenticate again. When the user visits the anchor's interactive flow URL and provides a `jwt`, the anchor must verify the JWT, and if valid, honor the user's authenticated session. In this case, the anchor should immediately respond with a `302 redirect` response that sets a cookie authenticating the user and redirects the user to a different URL that does not contain the `jwt` parameter. This is to avoid the JWT appearing in the user's browser history. After the redirect, the authenticated interactive flow must continue as usual.
+
+#### Completing an interactive withdrawal as the wallet
+
+In the case of an interactive withdrawal, the user will interact with the anchor via the popup, entering information needed to complete the withdrawal like destination bank account or KYC details. Once the anchor has enough information to know how to complete the withdrawal, the wallet detect this and allow the user to complete the withdrawal inside the wallet's app. This is because a withdrawal is only properly initiated once the wallet transfers the correct amount of the withdrawal asset to the anchor's Stellar account.
+
+To do this, the wallet needs to either poll the `/transaction` endpoint with the `id` provided in the `interactive_customer_info_needed` response from the anchor until the necessary information is available, or register a callback with the anchor as described above. Either way, the anchor's response will contain the transaction fields described in the [/transactions](#transaction-history) endpoint.
+
+The wallet must use the response fields in the following way to complete the withdrawal:
+
+- `status`: `incomplete` means the user is either still going through the interactive flow with the issuer, or has abandoned the transaction part way through. The wallet should simply keep polling or waiting in this case. The wallet should provide the user with an `x` or other way to abort the transaction and return to their wallet in this case.
```

Comment:
Should we provide a way for the Wallet to inform the Anchor that the transaction has been aborted by the user? Or should it be the Anchor's responsibility to define the criteria for timing out a deposit / withdrawal?

Reply:
Good question. If the wallet closes the anchor popup when the `x` is clicked, then there's no need. The case that concerns me is where the user clicks the `x` in the wallet and then finishes the anchor flow. They'd end up back at the wallet trying to complete the transaction, but the wallet would not have the right interface up. That can't happen with the wallet closing the popup, though. So I think that's the best approach.

Reply:
Agreed!